backend/docs/devices_v1_deployments.oai.yaml

```diff
@@ -0,0 +1,396 @@
+        All parameters are generated internally when fetching a configuration deployment.
+      tags:
+        - Device API
+components:
+  responses:
+    NotFoundError:
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+      description: Not Found.
+    InternalServerError:
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+      description: Internal Server Error.
+    InvalidRequestError:
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+      description: Invalid Request.
+  schemas:
+    Error:
+      description: Error descriptor.
+      example:
+        error: "failed to decode device group data: JSON payload is empty"
+        request_id: f7881e82-0492-49fb-b459-795654e7188a
+      properties:
+        error:
+          description: Description of the error.
+          type: string
+        request_id:
+          description: Request ID (same as in X-MEN-RequestID header).
+          type: string
+      type: object
+    DeploymentStatus:
+      example:
+        status: success
+      properties:
+        status:
+          enum:
+            - installing
+            - pause_before_installing
+            - downloading
+            - pause_before_rebooting
+            - rebooting
+            - pause_before_committing
+            - success
+            - failure
+            - already-installed
+          type: string
+        substate:
+          description: Additional state information
+          type: string
+      required:
+        - status
+      type: object
+    DeploymentInstructions:
+      example:
+        id: w81s4fae-7dec-11d0-a765-00a0c91e6bf6
+        artifact:
+          artifact_name: my-app-0.1
+          source:
+            uri: https://aws.myupdatebucket.com/image123
+            expire: 2016-03-11T13:03:17.063493443Z
+          device_types_compatible:
+            - rspi
+            - rspi2
+            - rspi0
+      properties:
+        id:
+          description: Deployment ID
+          type: string
+        artifact:
+          $ref: "#/components/schemas/DeploymentInstructions_artifact"
+      required:
+        - artifact
+        - id
+      type: object
+    DeploymentLog:
+      example:
+        messages:
+          - timestamp: 2016-03-11T13:03:17.063493443Z
+            level: INFO
+            message: OK
+          - timestamp: 2016-03-11T13:03:18.023765782Z
+            level: DEBUG
+            message: successfully updated.
+      properties:
+        messages:
+          description: Array of log entries of a deployment
+          items:
+            $ref: "#/components/schemas/DeploymentLog_messages_inner"
+          type: array
+      required:
+        - messages
+      type: object
+    DeploymentInstructions_artifact_source:
```

Comment:
was this your decision to introduce those, I mean: to separate every object into a new schema definition? also: I will not comment about it more, I do not think we were using this name notation before, was it a decision to switch to it?